src/statefulmodelinstance.cpp

```diff
@@ -180,117 +180,84 @@ const Status StatefulModelInstance::validateSpecialKeys(const tensorflow::servin
     return StatusCode::OK;
 }
 
-template <typename RequestType>
-const Status StatefulModelInstance::validate(const RequestType* request, SequenceProcessingSpec& sequenceProcessingSpec) {
-    OVMS_PROFILE_FUNCTION();
-    auto status = validateSpecialKeys(request, sequenceProcessingSpec);
-    if (!status.ok())
-        return status;
-
-    return request_validation_utils::validate(
-        *request,
-        getInputsInfo(),
-        getName(),
-        getVersion(),
-        SPECIAL_INPUT_NAMES,
-        getModelConfig().getBatchingMode(),
-        getModelConfig().getShapes());
+const std::set<std::string>& StatefulModelInstance::getOptionalInputNames() {
+    return SPECIAL_INPUT_NAMES;
 }
-
-Status StatefulModelInstance::infer(const tensorflow::serving::PredictRequest* requestProto,
-    tensorflow::serving::PredictResponse* responseProto,
-    std::unique_ptr<ModelInstanceUnloadGuard>& modelUnloadGuardPtr) {
+template <>
+StatefulRequestProcessor<tensorflow::serving::PredictRequest, tensorflow::serving::PredictResponse>::StatefulRequestProcessor(SequenceManager& sequenceManager) :
+    sequenceManager(sequenceManager) {
+}
+template <>
+Status StatefulRequestProcessor<tensorflow::serving::PredictRequest, tensorflow::serving::PredictResponse>::extractRequestParameters(const tensorflow::serving::PredictRequest* request) {
     OVMS_PROFILE_FUNCTION();
-    enum : unsigned int {
-        GET_INFER_REQUEST,
-        PREPROCESS,
-        DESERIALIZE,
-        PREDICTION,
-        SERIALIZE,
-        POSTPROCESS,
-        TIMER_END
-    };
-    Timer<TIMER_END> timer;
-    using std::chrono::microseconds;
-    SequenceProcessingSpec sequenceProcessingSpec;
-    auto status = validate(requestProto, sequenceProcessingSpec);
-    if (!status.ok())
-        return status;
-
-    std::unique_lock<std::mutex> sequenceManagerLock(sequenceManager->getMutex());
-    status = sequenceManager->processRequestedSpec(sequenceProcessingSpec);
+    auto status = StatefulModelInstance::validateSpecialKeys(request, sequenceProcessingSpec);
```

Comment:
`extractRequestParameters` name might be misleading, for non stateful it does nothing which is ok but for stateful it valides the keys instead of extracting, consider changing the name to reflect behavior

Reply:
Actually it extracs the sequenceID & sequence control input & validates at the same time. For me it just could fail during extract so this is the least misleading name I could come with. I'm open for name suggestions.

Reply:
Agreed to rename validateSpecialKeys -> extractSpecialKeys